.github/workflows/golangci.yml

```diff
@@ -19,8 +20,18 @@ jobs:
         with:
           go-version: 1.19
       - uses: actions/checkout@v3
+      # On pushes and on pull requests from branches that target main, run all linters.
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.4.0
+        if: github.base_ref == 'main' || github.event_name == 'push'
```

Comment:
What is the `|| github.event_name == 'push'` for?

Reply:
`base_ref` is only available when the event is `pull_request`, the OR allows it to run even on direct pushes (which I believe we never do but I didn't tweak the already existing condition here since I was unsure).